### PR TITLE
Box Beautification & Bar Rework & Issue 9706 Fix

### PR DESCRIPTION
**Pull request by KathrinBailey**
## About The Pull Request
A facelift for Box Station, many minor changes that stack up to making a better BoxStation©.
Bar and dorms rework with minor changes and beautifications in other places. Adds gas miners also.

**I'm clueless about the map merger stuff and haven't been able to run it (says it needs a module named bidict). Map merger has not been run on the map, so that may be a problem. If I need to do stuff with this then it's probably easier to give someone the map file for the mto run the merger or something.**

## Why It's Good For The Game
Makes the map more beautiful in general. Brings up Box's very old bar that has had no love at all to modern standards. The bar is now closet to a restaurant, it looks much nicer and from the pictures I've put in Discord, people seem to like it. Expands dorms with two 'luxury' rooms that have their own bathroom. And before anyone asks: **No functionality has been lost when making space for the changes. Everything has just been nudged slightly.** This doesn't make Box any 'bigger' either.
What you can do if this gets pushed:
• Make your own space restaurant! You can now, as a gimmick, cooperate as a department to make a restaurant.
• Have enough table space in the bar to seat a group up to 6 people!
• Enjoy downtime in a group in two large luxury cabins!
• Make a fun gimmick room in maint!
• Lock the doors whilst on the toilet, so that pesky borg doesn't barge in!
Anyway, the primary purpose of this PR is to improve the standard of roleplay in the server and to make the map look prettier.

## Changelog
:cl:
add: Two extra 'luxury' dorms rooms!
add: Gas miners to atmos.
add: Posters around the station.
add: Vacant room added to the Starboard Bow with it's own APC, above electrical maintenance.
add: New trendy clothes to the locker room, giving variety and bringing fashion back onto Nanotrasen stations.
add: Coloured bedsheet and towel bin.
add: Maid uniforms for the janitor.
tweak: Completely reworked bar. Milk kegs added in bar office. The bar has been changed for a homey restaurant feel just in time for Christmas! You can now run it as an actual restaurant! Local Bartender Icktsie III loved it so much he rolled around on the new floor smiling happily.
tweak: Dorms rework. Fitness room now has lots of costumes and outfits.
tweak: Junk removed from engineering, welding goggles added.
tweak: Welding tools in engineering replaced with industrial welding tools.
tweak: Package wrappers and hand labellers now in major departments.
tweak: Cell charger moved from engineering lobby to the protolathe room, just like how it is in all of the other maps and just where the cell charger is actually needed.
tweak: Library redesigned to have a private room and a 3x3 private study that is cleaned up.
tweak: Paper bins have gone big or gone home, with premium stationery scattered around. Engineering and security now have a labeller and packaging supplies.
fix: Dark spot top left of Botany fixed.
fix: Huge galactic-sized dark spot in bar fixed.
fix: Light replacers now are less horrifically overpowered and PTSD-inducing for the server.
fix: Fixes issue 9706: https://github.com/Citadel-Station-13/Citadel-Station-13/issues/9706 Part of maint getting hit by radstorms.

_Kathrin's Box Beautification:_
/:cl:

Original changelog (other changes in comments):
**Primary Changes**
• Complete bar rework, featuring a restaurant-like theme. Tablespace increased by 140% with seats with differing capacities to support small and large groups. Fixes the _**huge**_ dark spot in the bar with a central 'pillar' with lights all around. Bar now has wooden flooring all over except behind the bar. There's also a condiments table. Fluff notes and a paper bin have been left behind, there's a noticeboard all to facilitate a restaurant gimmick, should hopefully raise the RP level. One arcade machine of every type added.
• Bar Office cleaned up, the shotgun has been moved to the ID-locked cabinet. Getting a free shotgun round-start with only a welder and wrench is a little odd in low-pop when there is no Bartender. Coffee and cola vending machines removed, the Bartender is next to a giant cola dispenser anyway, kegs are in their place. Waiter uniforms added.
• Bathroom rework, it's now compressed and has bolt buttons for the doors. Each restroom now has a sink, mirror, toilet, bolt switch and light in the same tile, much closer to a real-life restroom. You cannot see through the shower curtain when it is closed now! You also can't see through it when it is open, so that's a trade-off. It was either you can always see or you can't. Privacy is good. The floor is now blue titanium, it has a more bathroomy feel.
• Added two new 'premium' dorms rooms with their own bathrooms. They both have a fireplace and storage.
• Increased the size of the two 'Cabin' rooms from a 3x2 to a 3x3, removed the table and stool on the other old dorm rooms, they just take up space and offer no function.
• Fitness room reworked. It now is a secondary locker room except without the personal closets. This means future mappers can remove/rework the old one if they want. This is because it is next to the holodeck, no one wants to put on holographic clothes, walk out and be naked, or to go into your holo-adventure dressed as an engineer. To make space for this, maint right of dorms has been compressed with no loss of functionality or feature whatsoever.
• All dorms rooms now have numbers and renamed doors to a consistent naming scheme. A washing machine and linen bin has been added
• Removes theatre. No one uses it.

**Secondary Changes**
• Added a small dorm room under the locker room, compressing the locker room showers to save space. No one uses these showers, so this room gives people an opportunity for privacy.
• Added an 'Instrument Storage' room in maint, using existing space. It has instruments and a spare piano, this is done to facilitate more gimmick opportunities.
• Added a new room into maint by deleting unused space to comply with a Discord request for a 'DIY' room in maint. It's at the top right and small.
• Posters added throughout the station, such as a PDA ad and Love Ian posters around the HoP office, cigarette posters in the Detective's office, cola brand posters next to their respective vendors' etcetera. This should add a bit of colour and life to the station.
• Locker Room now has a linen bin, it didn't before for some reason.
• Maid uniforms added to janitor closet. Posters added to janitor closet.
• Added gas miners into atmospherics gas tanks.
• Added two firefoam and high-cap water dispensers into atmos because why not?
• Arrivals now has some catwalks and lattices in space with big bright anchored docking lights that I have on my own map. It just looks pretty.
• Adds a decorative bridge in space to the south of the station to make EVA a little less consistent.

first pr ever please don't bully

##Images:
https://i.gyazo.com/ef83f7ec616ff5acbdd7c8d549055f0d.png
https://i.gyazo.com/bd4c0e4d1fdb7a4705c91d4630e1e5a4.png
https://i.gyazo.com/cf273a97cb9d64a5d5798dd61e4e4f02.png
https://i.gyazo.com/f9a420340fca30349e49ea94d107a0fd.png
https://i.gyazo.com/48369f96429d6a7b5490e76734c19ac1.png
https://i.gyazo.com/474370cb2949a57c815b2464a1d98bfe.png (Miners added to other tanks too)
https://i.gyazo.com/51f01f9c4f4a2b3203d0fa8932affed6.png 
https://i.gyazo.com/98c775f3dcd764bb62e803ebe8ea5cb8.png (overview pic)

### CODE REVIEW

**Comment by r4d6** [comment]
Can you post screenshots on the description? you can change it

**Comment by KathrinBailey** [comment]
> Can you post screenshots on the description? you can change it

Pics added.

**Review by deathride58 — DISMISSED** [review]
Use the mapmerger.

**Comment by Ghommie** [comment]
Alright so, time to give this one a little review. Well, wall of text.

You seem to have offset quite a few lockers, drawers and fireplaces in dorms. Stylishness aside, they will indeed look off when placed beside non-offset counterparts, also one of the fireplaces seems to overlap over the APC found right behind it, a little bit of an issue for players able to see through walls.
Please undo the offset any other coding program of your choice, but **not** dreammaker, because it will leave dirty var-edited `pixel_x = 0` and `pixel_y = 0`, which Travis will complain about.

Any chance you could move the boxing ring to maintenance, maybe that little empty room you just made above the chapel, instead of just deleting it?

I see a piece of paper on a wall turf by the bathroom. I can only assume it's a shabby attempt at making it visually look like a note pinned to a wall. please place a `/obj/structure/noticeboard` on its turf, it will handle the rest on its Initialize().
Also make the paper note an actual type added to `code/modules/paperwork/paper_premade.dm` instead of a varedited one, same goes to the bar one.

The new airlocks leading to the tabletop game backroom of the library should be of the same type of the entrace, ergo glass ones.

There is an assistant start landmark by arrival maintenance. That could be a minor potential issue if the assistant-start-without-maintenance-access config was ever re-enabled (which I doubt).

**Comment by KathrinBailey** [comment]
> Alright so, time to give this one a little review. Well, wall of text.
> 
> You seem to have offset quite a few lockers, drawers and fireplaces in dorms. Stylishness aside, they will indeed look off when placed beside non-offset counterparts, also one of the fireplaces seems to overlap over the APC found right behind it, a little bit of an issue for players able to see through walls.
> Please undo the offset any other coding program of your choice, but **not** dreammaker, because it will leave dirty var-edited `pixel_x = 0` and `pixel_y = 0`, which Travis will complain about.
> 
> Any chance you could move the boxing ring to maintenance, maybe that little empty room you just made above the chapel, instead of just deleting it?
> 
> I see a piece of paper on a wall turf by the bathroom. I can only assume it's a shabby attempt at making it visually look like a note pinned to a wall. please place a `/obj/structure/noticeboard` on its turf, it will handle the rest on its Initialize().
> Also make the paper note an actual type added to `code/modules/paperwork/paper_premade.dm` instead of a varedited one, same goes to the bar one.
> 
> The new airlocks leading to the tabletop game backroom of the library should be of the same type of the entrace, ergo glass ones.
> 
> There is an assistant start landmark by arrival maintenance. That could be a minor potential issue if the assistant-start-without-maintenance-access config was ever re-enabled (which I doubt).

The library doors were not the glass ones because they should have glass to var=0, the reason being that the adjacent windows have privacy shutters to make that room into an actual private study area, similar to Delta. I could put a button with curator and security ID just outside the room if there are concerns with it being too high secure or anything. The reason for privacy is so you can actually play D&D or something without the fear of people barging in through those doors. They can still barge in through maint since that doesn't get bolted, because it's the one fire exit.

The boxing ring I've never seen used, not only in my entire time on Citadel, but my entire time on SS13 as a whole, it seems so pointless since I see people using the holodeck to roleplay their Darth Maul vs Darth Maul fantasies. The new room in maint was requested as a 'DIY' room, the purpose being that it is just a small area that you can claim (hence the airlock electronic over the door) to create your own gimmick inside. It seems like a much better idea to just put the sports lockers near the holodeck and instead, if people actually want to have an open boxing ring, they could just grab some glass, rods and a screwdriver and make one in a minute flat. If people want to run illegal boxing matches and the likes, then that's one of the purposes of the DIY room, because you can build your own in there. Personally I'd argue that the theatre has twice the right to exist as the pre-built boxing ring. I'm open to the idea of using turf decals to make a 4x4 grid where a diagonal half being red and another being blue though, since if you build a ring around it, then you can actually have a red and blue corner. (sorry for wall of text)

Everything else I can do and I'll make a commit once it is done. The pieces of paper were essentially just fluff, it might be better to put the paper in the bartender's office instead of a noticeboard, all it does is say that the bar has been changed to facilitate a restaurant gimmick and that they should try running the bar as a restaurant. The same note is on a table in the Bartender's office on my map too.

-edit: its 4am grammar is not a priority

**Comment by Ghommie** [comment]
I must insist on the boxing ring. It's the only place where I can violently click other spessmen until they go horizontal for no reason, sort of. Maybe you were just too busy observing other events elsewhere, maybe because box's ring is the shittiest one, maybe because boxing needs a little overhaul...

**Comment by KathrinBailey** [comment]
Alright, everything should be good now.
If a boxing ring is seriously seen as necessary then you're going to have to be okay with me rebuilding a part of maint, probably the space tiles with the airlock painter above public botany/garden and tool storage so I can fit it in, because I may as well make it either good or let people build a ring themselves.

**Comment by Weblure** [comment]
This new layout is very open and refreshing, I like it a lot. I've spotted a few flaws during the testmerge, and I'd also like to suggest some opinionated changes. The following list aside, great work! I hope this PR gets fully merged in the future.

# Issues

**The borg charger in security maint has no area assigned to it, rendering it useless.**
image

**A lot of upwards-facing sofas aren't connected.**
image
image
image

**There are a couple of random red tiles in the bar.**
image

**This air alarm is obstructed in the public dorm bathroom.**
image

# Opinions

**Too Many Offsets**
Keep in mind that if anything gets damaged and needs to be replaced, the replacement wont receive your custom offset. Also, applying offsets to movable items will either cause those items to be permanently offset while being dragged around (not good) or they'll just lose their offset when moved.

That being said, all of these locker and closet offsets are not ideal.
image
image

This intercom is mounted on a brick chimney.
image

This intercom is offset above an offset dresser. I get what you're going for, but I don't think it's necessary with all the free wall space you have to work with.
image

I'm a bit on the fence about these full, 1-tile bathrooms. Toilet sprites already have poor perspective, and offsetting the sprites so that they're flush with the walls kinda makes them look like they're sitting *on* the walls, as if this were a side-scrolling game. Also, a 1x1 room filled with both a toilet and a sink would probably be cramped and uncomfortable.
image

**The public showers should have a door instead of a curtain.** Having open curtains that you can't see through is janky, and curtains don't seem very fitting for a communal shower area. If you don't want to use an airlock, a wooden or metal door would work too.
image 

**This note is a bit unnecessary and would require another PR for its removal in the future, when the renovations are no longer considered 'new'.**
image

**The theatre backstage has wasted wall space.** I recommend moving the vendor up a tile, the door down a tile, moving the disposals to the left of the vendor, and then putting the second fridge back where it was. Refrigerators are very useful for clown shenanigans.
image

**Comment by R3dtail** [comment]
Some of the vending machines seem moved around for ...no real reason.  Like  the one in the locker room in arrivals.  Also I'm pretty agianst the idea of adding MORE dorms.  The amount we had were fine.  Also , removing the slot machine in the area above chapel is kinda lame, especially with all the coins around it.    The fact that the dorms are so needlessly large is pretty meh in my eyes, too.  Like, the dorms should've been left 3x2.  That offered just enough room for , really anything you could possibly want.  especially since most things involved not even looking at the room, but at the text box.  

Bar looks nice though, no complaints there.

**Comment by KathrinBailey** [comment]
Hi, thanks for all the feedback!
Definitely didn't spot a lot of these once testing, so thanks a lot!
Awkwardly, the sofas appear like this in the editor:
image

It seems like Persi had the same issue on the Halloween map, I'm switching the sofas around in the editor.
Looks like I actually forgot way more with the theatre backroom, the scrubber and disposals weren't connected.
RLD nerf part of the commit, 500 matter units to 200.

The new 'luxury' dorms were sure popular...
image
None of the other rooms was occupied!


For Redtail:
I don't see a problem with the vending machines in the locker room. Consumables are on one side, clothes on the other, it's just for consistency.
The coins were never removed, they were just moved. The slot machine is gone, however, there is still the one in the bar.
No space was wasted with dorms, it is taking up no space that had any inkling of significance or purpose before, I have received nothing but positive feedback for the dorms and I have to disagree with the statement that 3x2 dorms are adequate. If you have a small group it just isn't adequate, plus, there was previously no private space for a small group to go with doors to bolt.
-edit: Also now one dorms bomber shouldn't vent every single room and being a barge-in shitter is less appealing.

Changes:
• Added instruments to instrument storage.
• Packaged a rapid-light-device nerf from 500 matter units capacity and loaded to 200 to stop spamming lights. Originally it was going to be 50, Aelita said 250, 200 seems okay for now. If it gets spammed and abused I will heavily nerf it to 50 or 100 capacity.
• Paper bin and pen in security office now all in one table, more donuts.
• On request of Forrest, the security EVA suits were moved from armoury to Warden's office, you still need armoury access (practically) to get to them, however, once the armoury is spaced from a bomb or thermite-d wall or other things you shouldn't have to walk into a vacuum to equip your vacuum-proof suit.
• All bugs stated above fixed (**Massive thanks to Weblure for so helpfully compiling them all!**)
• Theatre backstage fixed as per the suggestion and fixed because it was broken anyway.
• Sofas _should_ be fixed now.
• Intercoms in dorms rooms moved.
• Bathroom air alarm moved.
• AAHHH SPEED COMMIT OTHER STUFF I'LL EDIT IT INNN

I didn't change shower curtains yet, I thought it'd be worth asking and see if people prefer it over an airlock. Show curtains either never block line of sight or always block line of sight, being opened or closed does nothing. You can destroy the tinted windows that make this effect happen in-game if you /really/ want to.
-edit also will ask about if people prefer the visual benefit of the furniture being against walls more than the jankiness it may cause or not.

**Comment by KathrinBailey** [comment]
Reviews:
image
image
image
image
image

**Comment by KathrinBailey** [comment]
Fixes:
**Note: Some changes already have existed, but have not being commented on, so they are being listed so people are aware.**
image
Firelocks should behave better.

image
The last fix I deleted the two red decals, but I ctrl-z ed evidently one too many times and one was left over. Oops. Now both are gone.

image
Areas fixed, oops. Don't try and rush it for a commit I guess. Having the room run downwards like this gives one extra tile back to maint.

image
Although it doesn't appear so in the editor, that pixel offset PR should make this look neater. Towel and coloured bedsheets, before it was just plain white bedsheets. Coloured bedsheets added to locker room.

image
I didn't ever expect that anyone could mistake this room for part of dorms, but since apparently that is a concern, it has been replaced with... 'Furniture Storage'. The suits that were in a closet have now had the closet repurposed as a suit closet. It now has lots of suits.

image
Gas miners. This already existed, but I just want to make anyone who hasn't noticed aware that they've been added.

image
Consistency and a general clean-up. All other maps position the cell charger near the protolathes. Box should not be an exception. Books from the top half of the image have moved to the old cell charger spot. In secure tech this are two RLDs. No one has expressed concern about these and they should now be nerfed to stop the spamming of lights, for performance reasons.
One silly thing that existed prior is the existence of normal welding tools in a place where all of the workers spawn with industrial welding tools. The welding tools on racks and tables have been upgraded to industrial ones. Some basic tools around engineering have been removed because they are completely redundant. Engineers already have a crowbar. If they are actually that desperate for one and have somehow lost it before experimental tools have been researched, and also the autolathes have vanished, they can use one of the vendors. Spots in engineering that have welding helmets now have welding goggles, as eye protection in the glasses slot instead of the helmet slot isn't valuable in a department where seeing a shiny crystal makes you hallucinate for the next hour.

image
As requested, the two EVA suits have been moved two tiles down. The wall between the armoury and space is often bombed, for security to fix it, they need space-proof suits. Having these space-proof suits be in a vacuum doesn't make sense. The Warden's office has been tidied up to fit the suits. The suits also still require armoury access to get to. The Warden has armoury access, and I doubt any HoP is giving Warden's Office access but not Armoury access if that's even possible.

image
Library private room, similar to what exists on Meta. Yes, the airlocks are different. Having glass airlocks defeats the purpose of a 'private' room. The windows have shutters and the doors (despite as they appear in the image) have no glass and can be bolted. The maint door cannot be bolted and a button to unbolt it is outside the Curator's office, only the Curator and Security can use this button to unbolt, whereas anyone can use the button to unbolt in the private room itself.

image
Final version of the bathroom. On request, the showers have been made 3x3 again and some people prefer airlocks over the curtain, so there is an airlock again.

image
Final version of the bar. Yes, the sofas appear fucked in the editor, but apparently it appears normal in game. By (an extremely well-received) request, a keg of milk is in the Bartender's office. Not much else I can say about it that hasn't been seen already. Here is the old box bar.
image
I'm not sure how many years that massive giant dark spot has been there, but hopefully, it'll finally be gone soon.

image
Apparently the coins that were moved were too hard to find, so hard that people thought I removed them, so I have placed more and some are in more obvious locations. The slot machine in top right maints has been placed back.
A small room has been added on request with its own area and APC. If people want to build a gimmick room, they can do so here.

The boxing ring issue has been """"resolved"""". In other words, there's space for one, turf_decals around to make it look nice and fancy. A red corner and a blue corner. There is no prebuilt ring as it takes far too much space to justify its existence, considering it is almost never used. If you're that desperate to stamcrit your friends that you don't want to spend two minutes making a ring, the holodeck is also two tiles away.

That's every change of any significance listed either here or above, every known bug should now be fixed and this should be the final commit for this PR. 
I was requested to redesign the bathroom and make it bigger, however, if I do so, I'm expecting that someone, somewhere, will take massive issue with it and then it won't be merged until next year


_This is probably full of grammar and spelling errors it's 9am I just want to go to bed.._

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -663,8 +663,8 @@ RLD
 	icon_state = "rld"
 	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
-	matter = 500
-	max_matter = 500
+	matter = 200
```
Comment (Sishen1542) [inline code/game/objects/items/RCD.dm]: ????

Comment (KathrinBailey) [reply code/game/objects/items/RCD.dm]: You didn't copy and paste this correctly.

The RLD is overpowered and needs nerfing for it to be viable to be used in-game, it probably should be nerfed much harder than this.

Comment (Sishen1542) [reply code/game/objects/items/RCD.dm]: You're doing something completely unrelated to the map changes... in a map change PR. 
That's not how it works.

**Comment by JTGSZ** [comment]
Nice job on that unlisted nerf in a map rework

File: code/game/objects/items/RCD.dm [hunk 2]
```diff
@@ -663,8 +663,8 @@ RLD
 	icon_state = "rld"
 	lefthand_file = 'icons/mob/inhands/equipment/tools_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/tools_righthand.dmi'
-	matter = 500
-	max_matter = 500
```
Comment (JTGSZ) [inline code/game/objects/items/RCD.dm]: Hello, Make a separate PR for this.

**Review by JTGSZ — CHANGES_REQUESTED** [review]

**Comment by ForrestWick** [comment]
Finally, box is worth playing. And that maint dorm would be great for antg shit

**Comment by ShadeAware** [comment]
It wouldn't be a stretch to say this PR makes Box the best map in the game atm 👌

**Comment by Knouli** [comment]
Medbay of the new box is lacking its roundstart bloodbank generator. This has become a standard across all maps that we now include in rotation.

Here we see it in old boxstation pre testmerge.
image

Here it is on Metastation.
image

Here it is even on Omegastation.
image

Here in Pubby as well.
image

And yet on the testmerge map we're lacking a bloodbank.
image

Please add this in! It is a very useful component to medbay's arsenal, and something that's been decisively made a round start feature across every single other map we play on! The re 'X' is where I would personally replace the bloodbank, it's exactly where it was on pre testmerge box and I never encountered any issue with its placement there, just shunt the displaced boxes over to the L table along the bottom wall.

**Comment by Knouli** [comment]
I see that the blood bank machine is now available from spawn! And was very likely a fix made before I was able to comment of it . . . Very good! Thank you!

I am, however curious of the new position of the blood machine and what seems to be my relative inability to count both syringe guns at round start. I've either been part of a couple shifts where the doctors were particularly grabby on both of them, or the placement of the bloodbank has knocked off the secondary spawn that pops up within the equipment room. Though, I am able to at least confirm the presence of the one in patient recovery.

**Comment by GrayRachnid** [comment]
HoP office missing stamp

**Comment by GrayRachnid** [comment]
The 2 new dorm cameras need to be renamed to match the original 2. Right now we have Dorms South, Dorms, Fitness Room, and Holodeck Costumes. The last 2 need to match the first 2.

**Comment by GrayRachnid** [comment]
Det office missing cuffs.

**Review by silicons — APPROVED** [review]

**Comment by AffectedArc07** [comment]
For future reference, `pip install bidict`  or `python -m pip install bidict` depending on how you have python installed
